src/transpile/cacheblocking.hpp

```diff
@@ -0,0 +1,684 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _aer_cache_blocking_hpp_
+#define _aer_cache_blocking_hpp_
+
+#include "transpile/circuitopt.hpp"
+#include "framework/utils.hpp"
+
+
+namespace AER {
+namespace Transpile {
+
+class CacheBlocking : public CircuitOptimization {
```

Comment:
I think it would be very useful to have an actual example in comments on how a circuit changes when this optimizer is applied. Something like circuit before and circuit after.